File: rubysnippets/sieveOfErasthothenes.py


class Eratosthenes(object):

    def find(self, toSift):
        i = 0
        while i < len(toSift):
            u = i + 1
            while u < len(toSift):
                if toSift[u] % toSift[i] == 0:
                    toSift.pop(u)
                else:
                    u += 1
            i += 1
        return toSift

if __name__ == "__main__":
    e = Eratosthenes()
    print e.find(range(2,30))

File: golangsnippets/sieveOfErasthothenes.go
require 'set'
require './nextPrime.rb'
require './primeFactor.rb'


class Eratosthenes

  def find(toSift, num)
    init_num = 2
    while init_num < num
      toSift.delete(init_num)
      init_num += 

  end

end
print "Enter a number: "
num = gets.chomp.to_i
e = Eratosthenes.new()
puts e.find((2..num).to_a.to_set, num)
